doc/modules/model_persistence.rst

```diff
@@ -86,7 +87,40 @@ same range as before.
 Since a model internal representation may be different on two different
 architectures, dumping a model on one architecture and loading it on
 another architecture is not supported.
```

Comment:
Actually I'm not sure this sentence is correct https://github.com/scikit-learn/scikit-learn/pull/17644#issuecomment-661896033 As long as the Python version and the versions of dependencies are the same it should work. When it doesn't it could be a bug on our side, particularly given that pickle (+ docker) is still the primary method of deployment.

Also (sentence below) containers are architecture specific. They are a solution to reliable deployments and fixing the environment including the dependencies, but not to architecture portability.

Reply:
> Actually I'm not sure this sentence is correct [#17644 (comment)](https://github.com/scikit-learn/scikit-learn/pull/17644#issuecomment-661896033) As long as the Python version and the versions of dependencies are the same it should work. When it doesn't it could be a bug on our side, particularly given that pickle (+ docker) is still the primary method of deployment.

I've tried to mitigate the sentence: but the point here is more if scikit-learn developers are willing to guarantee pickle portability than about pickle portability itself.

> Also (sentence below) containers are architecture specific. They are a solution to reliable deployments and fixing the environment including the dependencies, but not to architecture portability.

Again, I've tried to mitigate the sentence, but IMO any solution enforcing portability and interoperability is architecture dependent, the goal is to make this dependence invisible to the user. The "pickle + docker" option offers to the user a way to run predictions with a model trained on a different architecture... obviously the producer of the model should provide the right "packaging" to the user.